BulgarskiChiflikGuestHouse.DAL/Repository/Repository.cs

```diff
@@ -25,14 +25,19 @@ public async Task<List<Room>> GetAllRoomsAsync()
 
         public async Task<Room> GetRoomByIdAsync(Guid roomId)
         {
-            var room = await _guestHouseDbContext.Rooms.FirstOrDefaultAsync(r => r.Id == roomId);
+            var room = await _guestHouseDbContext.Rooms
+                .Include(room => room.ImagesSources)
+                .Include(room => room.RoomType)
```

Comment:
This style of multiple relationships includes can contributes to weak performance use .ThenInclude() take a look at the link ->
https://learn.microsoft.com/en-us/ef/core/querying/related-data/eager